src/cpp/rtps/builtin/data/ParticipantProxyData.cpp

```diff
@@ -391,17 +397,15 @@ void ParticipantProxyData::clear()
     m_userData.clear();
 }
 
-void ParticipantProxyData::copy(ParticipantProxyData& pdata)
+void ParticipantProxyData::copy(const ParticipantProxyData& pdata)
 {
     m_protocolVersion = pdata.m_protocolVersion;
     m_guid = pdata.m_guid;
     m_VendorId[0] = pdata.m_VendorId[0];
     m_VendorId[1] = pdata.m_VendorId[1];
     m_availableBuiltinEndpoints = pdata.m_availableBuiltinEndpoints;
-    m_metatrafficUnicastLocatorList = pdata.m_metatrafficUnicastLocatorList;
-    m_metatrafficMulticastLocatorList = pdata.m_metatrafficMulticastLocatorList;
-    m_defaultUnicastLocatorList = pdata.m_defaultUnicastLocatorList;
-    m_defaultMulticastLocatorList = pdata.m_defaultMulticastLocatorList;
+    metatraffic_locators = pdata.metatraffic_locators;
+    default_locators = pdata.default_locators;
```

Comment:
We have to study if m_readers and m_writers attributes have to be copied too.

Reply:
This method is only called when a new participant is discovered. The destination of the copy will always be a new ParticipantProxyData or one from the pool, so there is no need for m_readers and m_writers to be copied

Reply:
Added comments on commit 8500a0c